pkg/apis/policy/validation/validation.go

```diff
@@ -50,8 +50,20 @@ func ValidatePodDisruptionBudgetUpdate(pdb, oldPdb *policy.PodDisruptionBudget)
 func ValidatePodDisruptionBudgetSpec(spec policy.PodDisruptionBudgetSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	allErrs = append(allErrs, extensionsvalidation.ValidatePositiveIntOrPercent(spec.MinAvailable, fldPath.Child("minAvailable"))...)
-	allErrs = append(allErrs, extensionsvalidation.IsNotMoreThan100Percent(spec.MinAvailable, fldPath.Child("minAvailable"))...)
+	if spec.MinAvailable != nil && spec.MaxUnavailable != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, spec, "MinAvailable and MaxUnavailable cannot be both set"))
+	}
+
```

Comment:
I think that a PDB with a MaxUnavailable of 0 is a valid instance of an API object. As much as we might want all users to avoid this, and to design their infrastructure such that it will remain available during node failures (planned or unplanned), I don't think validation is the right place to enforce this.
If k8s users want to block all drains for critical production infrastructure such that the cluster and application administrator(s) have to coordinate a planned outage during a maintenance window, I think we should allow them to do so (while encouraging them to come up with a better strategy in the long term).